staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -546,15 +683,15 @@ func watchHandler(start time.Time,
 	name string,
 	expectedTypeName string,
 	setLastSyncResourceVersion func(string),
+	shouldExitPredicate func(metav1.Object) bool,
```

Comment:
I'm not a fan of it.

I think I have two concerns here:
1) I don't think we should exit the watch after receiving the bookmark - we should just pass callback that will replace the underlying store, but will continue to read further data from this watch.

2) I don't think we need to be so generic here. I would personally add make it usecase-specific, so basically
switch it to be bookmark-specific thing.

So I would like this to look more like
1. here:
   `onInitialEventsEndBookmark func()`
2. in the code itself I would rather go with the following:

```
   ...
   case watch.Bookmark:
     if isInitialEventEndBookmark(...) {
        onInitialEventEndBookmark(store)
     }
   default:
     ...
}
...
```

the `onInitialEventsEnd` function would be doing something like:

```
   store.syncWith(}the internal store items}
   // set RV etc.
```
[the internal store will effectively be part of the fuction when creating this callback so we can have access to it]

Reply:
So given our short conversation on slack, I wrote down (non-compiling) poc how I would envision this:
https://github.com/kubernetes/kubernetes/compare/master...wojtek-t:kubernetes:reflector_poc?expand=1